### PR TITLE
Lower bandwidth at startup

### PR DESCRIPTION
**Pull request by paulrd**
Before submitting a PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [x] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (run `lein do clean, test`)
- [x] Code inlining with mranderson works and tests pass with inlined code (run `./build.sh install` -- takes a long time)
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [x] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

### CODE REVIEW

**Comment by paulrd** [comment]
This addresses #263

File: src/refactor_nrepl/artifacts.clj [hunk 1]
```diff
@@ -10,10 +10,23 @@
             [refactor-nrepl.ns.slam.hound.regrow :as slamhound-regrow]
             [version-clj.core :as versions])
   (:import java.util.Date
+           java.util.zip.GZIPInputStream
            java.util.jar.JarFile))
 
+(def artifacts-file ".artifacts-cache")
```
Comment (expez) [inline src/refactor_nrepl/artifacts.clj]: I think we should move this to the platform's temp dir:

1. That way we can share it across all projects, independently of the dir the REPL was started in
2. It makes it unnecessary to add this file to the `.gitignore` of every project you work on or your git config.

File: .gitignore [hunk 2]
```diff
@@ -24,4 +24,5 @@ install.cmd
 /.cljs_node_repl/
 /out/
 /.lein-env
-.inline-deps
\ No newline at end of file
+.inline-deps
+/.artifacts-cache
```
Comment (expez) [inline .gitignore]: Should probably add a small note to the changelog too.

**Comment by paulrd** [comment]
Cool I'll make those changes. I'm having a hard time testing this in emacs, how do you go testing this within emacs?

**Comment by paulrd** [comment]
runnig ./build install i get the following:
Exception in thread "main" java.lang.ClassNotFoundException: javax.xml.bind.DatatypeConverter, compiling:(cljs/util.cljc:1:1)

**Comment by paulrd** [comment]
I think I know my issue with build.sh, I keep forgetting about the proper profile.

**Comment by paulrd** [comment]
Testing in emacs is giving me different behavior. It's downloading on every invocation of jack-in. It is getting the gzipped version at least :-).

**Comment by paulrd** [comment]
Is make inline-deps always required before I do lein install?

**Comment by paulrd** [comment]
ah! :force true is set! why!?

**Comment by paulrd** [comment]
Ah, ok it's not a bug. The default custom variable cljr-populate-artifact-cache-on-startup is true. Changing it to nil fixes the issue. I might open an issue there just as a reminder.

**Comment by paulrd** [comment]
One more thing I should change I should either keep the map sorted or resort it when loading from cache file.

File: src/refactor_nrepl/artifacts.clj [hunk 3]
```diff
@@ -10,10 +10,24 @@
...
+(defn get-last-modified-from-file
+  "Returns last modified time in milliseconds or nil if file does not exist."
+  [file]
+  (let [lm (.lastModified (io/file file))]
+    (if (zero? lm) nil lm)))
+
 ;;  structure here is {"prismatic/schem" ["0.1.1" "0.2.0" ...]}
-(defonce artifacts (atom {} :meta {:last-modified nil}))
+(defonce artifacts (atom (if (.exists (io/as-file artifacts-file))
+                           (->> artifacts-file slurp read-string (into (sorted-map)))
```
Comment (expez) [inline src/refactor_nrepl/artifacts.clj]: We should be able to use `edn/read-string` as a drop-in replacement here and avoid exposing the full reader like this.  As you probably know `read-string` is capable of evaluating any code inserted into that file.

Comment (paulrd) [reply src/refactor_nrepl/artifacts.clj]: Haha my spidy senses tingled on that but for some reason I never really understood that that's one of the things edn brought to the table. Thanks!

Comment (expez) [reply src/refactor_nrepl/artifacts.clj]: You could also bind [\*read-eval\*](https://clojuredocs.org/clojure.core/*read-eval*) to `false`, but it reads way better to just use the EDN reader when possible :)

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**